x/concentrated-liquidity/lp_test.go

```diff
@@ -622,6 +622,326 @@ func (s *KeeperTestSuite) TestWithdrawPosition() {
+
+			var (
+				ctx                         = s.Ctx
+				concentratedLiquidityKeeper = s.App.ConcentratedLiquidityKeeper
+				owner                       = s.TestAccs[0]
+				tc                          = tc
+				config                      = *tc.setupConfig
+				sutConfigOverwrite          = *tc.sutConfigOverwrite
+				err                         error
+			)
+
+			// If specific configs are provided in the test case, overwrite the config with those values.
+			mergeConfigs(&config, &sutConfigOverwrite)
+
+			// If a setupConfig is provided, use it to create a pool and position.
+			pool := s.PrepareConcentratedPool()
+			fundCoins := sdk.NewCoins(sdk.NewCoin(ETH, config.amount0Desired), sdk.NewCoin(USDC, config.amount1Desired))
+			if tc.amount0ToAdd.IsPositive() && tc.amount1ToAdd.IsPositive() {
+				fundCoins = fundCoins.Add(sdk.NewCoins(sdk.NewCoin(ETH, tc.amount0ToAdd), sdk.NewCoin(USDC, tc.amount1ToAdd))...)
+			}
+			s.FundAcc(owner, fundCoins)
+
+			// Create a position from the parameters in the test case.
+			if tc.createLockLocked {
+				_, _, _, _, _, _, err = concentratedLiquidityKeeper.CreateFullRangePositionLocked(ctx, pool, owner, fundCoins, tc.timeElapsed)
+				s.Require().NoError(err)
+			} else if tc.createLockUnlocking {
+				_, _, _, _, _, _, err = concentratedLiquidityKeeper.CreateFullRangePositionUnlocking(ctx, pool, owner, fundCoins, tc.timeElapsed+time.Hour)
+				s.Require().NoError(err)
+			} else if tc.createLockUnlocked {
+				_, _, _, _, _, _, err = concentratedLiquidityKeeper.CreateFullRangePositionUnlocking(ctx, pool, owner, fundCoins, tc.timeElapsed-time.Hour)
+				s.Require().NoError(err)
+			} else {
+				_, _, _, _, _, err = concentratedLiquidityKeeper.CreatePosition(ctx, pool.GetId(), owner, config.amount0Desired, config.amount1Desired, sdk.ZeroInt(), sdk.ZeroInt(), DefaultLowerTick, DefaultUpperTick)
+				s.Require().NoError(err)
+			}
+
+			if !tc.lastPositionInPool {
+				s.FundAcc(s.TestAccs[1], fundCoins)
+				_, _, _, _, _, err = concentratedLiquidityKeeper.CreatePosition(ctx, pool.GetId(), s.TestAccs[1], config.amount0Desired, config.amount1Desired, sdk.ZeroInt(), sdk.ZeroInt(), DefaultLowerTick, DefaultUpperTick)
+				s.Require().NoError(err)
+			}
+
+			// --- System under test ---
+			newPosId, newAmt0, newAmt1, err := concentratedLiquidityKeeper.AddToPosition(ctx, owner, config.positionId, tc.amount0ToAdd, tc.amount1ToAdd)
+			if config.expectedError != nil {
+				s.Require().Error(err)
+				s.Require().Equal(sdk.Int{}, newAmt0)
+				s.Require().Equal(sdk.Int{}, newAmt1)
+				s.Require().Equal(uint64(0), newPosId)
+				s.Require().ErrorContains(err, config.expectedError.Error())
+				return
+			}
+
+			s.Require().NoError(err)
+			s.Require().Equal(config.amount0Expected.String(), newAmt0.String())
+			s.Require().Equal(config.amount1Expected.String(), newAmt1.String())
+
+			// We expect the position ID to be 3 since we have two setup positions
+			s.Require().Equal(uint64(3), newPosId)
```

Comment:
Please check the balances. That is, that `amount0Expected` and `amount1Expected` were withdrawn from the user and moved to the pool (not amounts provided e.g. `amount0ToAdd`)

Reply:
Added check here 710deaa